### PR TITLE
Changes the project to support multiple api groups

### PR DESCRIPTION
**Pull request by ahg-g**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Changes the project to support multiple api groups. This is needed to add a config group for ComponentConfig api.

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
do we want this in 0.1.0?

**Comment by ahg-g** [comment]
Yes, i think we may also want to change the domain to kueue.sh; so we would have core.kueue.sh and config.kueue.sh.

We actually own the kueue.sh domain :)

**Review by ArangoGutierrez — APPROVED** [review]
I don't see why not
/lgtm

**Comment by ahg-g** [comment]
The second commit changes the domain to `kueue.sh`.

**Comment by ahg-g** [comment]
the other option is to change the domain to `kueue.x-k8s.io` and `kueue` group to core

**Comment by ahg-g** [comment]
Removed the second commit.

**Review by ArangoGutierrez — APPROVED** [review]
LGTM was dropped after a change
/lgtm

File: PROJECT [hunk 1]
```diff
@@ -15,7 +16,6 @@ resources:
   version: v1alpha1
 - api:
     crdVersion: v1
-    namespaced: false
```
Comment (alculquicondor) [inline PROJECT:18]: was this unnecessary?

**Review by alculquicondor — COMMENTED** [review]
/lgtm

File: PROJECT (hunk 1 above)
Comment (ahg-g) [reply PROJECT:18]: the kubebuilder tool did it automatically, so basically the default is false.

**Pull request merged**